client/src/app/employee/employee-detail/state/employee.service.ts

```diff
@@ -0,0 +1,68 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { ToastrService } from 'ngx-toastr';
+import { throwError } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+import { EnvService } from 'src/app/shared/env.service';
+import { Employee, EmployeeUpdate } from './employee.model';
+import { EmployeeStore } from './employee.store';
+
+@Injectable({ providedIn: 'root' })
+export class EmployeeService {
+  readonly apiRootUrl: string;
+
+  constructor(
+    envService: EnvService,
+    private http: HttpClient,
+    private store: EmployeeStore,
+    private toastr: ToastrService,
+  ) {
+    this.apiRootUrl = envService.apiRootUrl;
+  }
+
+  getEmployee(id: string) {
+    this.store.setLoading(true);
+
+    this.http
+      .get<Employee>(`${this.apiRootUrl}/Employees/${id}`)
+      .pipe(
+        catchError((err) => {
+          this.toastr.error(`Could not get employee ${id}`);
+          return throwError(err);
+        }),
+      )
+      .subscribe({
+        next: (detail) => {
+          this.store.update((state) => detail);
```

Comment:
`update()` also takes a `Partial<T>` where `T` is the state.

This means we can pass `detail` directly:

```typescript
this.store.update(detail);
```

Reply:
👍

Reply:
fixed